### ISSUE TITLE
Tests suite failures

### ISSUE THREAD

**Issue by ghisvail** [ismrmrd/ismrmrd#36]
Am I the only one getting these results on Linux:

```
ghislain@imperial-lap:~/debian/packages/ismrmrd/build/tests$ ./test_ismrmrd 
Running 25 test cases...
Entering test suite "ISMRMRD Unit Tests"
Entering test suite "AcquisitionsTest"
Entering test case "test_acquisition_header"
Leaving test case "test_acquisition_header"; testing time: 216mks
Entering test case "test_acquisition_init_cleanup"
Leaving test case "test_acquisition_init_cleanup"; testing time: 92mks
Entering test case "test_acquisition_create_free"
/home/ghislain/debian/packages/ismrmrd/tests/test_acquisitions.cpp(63): error in "test_acquisition_create_free": check !cacqp->data failed
Leaving test case "test_acquisition_create_free"; testing time: 181mks
Entering test case "test_acquisition_copy"
Leaving test case "test_acquisition_copy"; testing time: 167mks
Entering test case "test_acquisition_make_consistent"
Leaving test case "test_acquisition_make_consistent"; testing time: 89mks
Leaving test suite "AcquisitionsTest"
Entering test suite "ImagesTest"
Entering test case "test_image_header"
Leaving test case "test_image_header"; testing time: 126mks
Entering test case "test_image_init_cleanup"
Leaving test case "test_image_init_cleanup"; testing time: 130mks
Entering test case "test_image_create_free"
/home/ghislain/debian/packages/ismrmrd/tests/test_images.cpp(63): error in "test_image_create_free": check !cimgp->data failed
Leaving test case "test_image_create_free"; testing time: 139mks
Entering test case "test_image_copy"
Leaving test case "test_image_copy"; testing time: 139mks
Entering test case "test_image_make_consistent"
Leaving test case "test_image_make_consistent"; testing time: 103mks
Leaving test suite "ImagesTest"
Entering test suite "NDArrayTest"
Entering test case "test_ndarray_init_cleanup"
Leaving test case "test_ndarray_init_cleanup"; testing time: 91mks
Entering test case "test_ndarray_create_free"
Leaving test case "test_ndarray_create_free"; testing time: 83mks
Entering test case "test_ndarray_copy"
Leaving test case "test_ndarray_copy"; testing time: 85mks
Leaving test suite "NDArrayTest"
Entering test suite "FlagTest"
Entering test case "test_is_flag_set"
Leaving test case "test_is_flag_set"; testing time: 189mks
Entering test case "test_set_flag"
Leaving test case "test_set_flag"; testing time: 178mks
Entering test case "test_set_flags"
Leaving test case "test_set_flags"; testing time: 73mks
Entering test case "test_clear_flag"
Leaving test case "test_clear_flag"; testing time: 169mks
Entering test case "test_clear_all_flags"
Leaving test case "test_clear_all_flags"; testing time: 74mks
Leaving test suite "FlagTest"
Entering test suite "ChannelTest"
Entering test case "test_is_channel_on"
Leaving test case "test_is_channel_on"; testing time: 1868mks
Entering test case "test_set_channel_on"
Leaving test case "test_set_channel_on"; testing time: 1771mks
Entering test case "test_set_channel_off"
Leaving test case "test_set_channel_off"; testing time: 1597mks
Entering test case "test_set_all_channels_off"
Leaving test case "test_set_all_channels_off"; testing time: 73mks
Leaving test suite "ChannelTest"
Entering test suite "QuaternionTest"
Entering test case "test_directions_to_quaternion"
Leaving test case "test_directions_to_quaternion"; testing time: 64mks
Entering test case "test_quaternion_to_directions"
Leaving test case "test_quaternion_to_directions"; testing time: 67mks
Entering test case "test_sign_of_directions"
Leaving test case "test_sign_of_directions"; testing time: 60mks
Leaving test suite "QuaternionTest"
Leaving test suite "ISMRMRD Unit Tests"

*** 2 failures detected in test suite "ISMRMRD Unit Tests"
```

Running on up-to-date Debian testing.

**Comment by naegelejd** [comment]
As far as I know yes. These tests all pass on OS X and Ubuntu 14. That being said, the actual failures aren't wrong. I will fix these and submit a PR.

I'm checking that the `data` and `traj` pointers inside an acquisition are set to `NULL` after calling `ismrmrd_free_acquisition`, which is undefined behavior because that function actually calls `free` on the acquisition. I copy-pasted the assertions from the `ismrmrd_cleanup_acquisition` test.

**Comment by ghisvail** [comment]
Thanks for looking at it. I'll apply your fix to the packaging of `v1.3.1`, unless you guys plan to make a patch release with the fix soon.
